react/src/components/ImageList.tsx

```diff
@@ -320,6 +363,57 @@ const ImageList: React.FC<{ style?: React.CSSProperties }> = ({ style }) => {
     },
   ];
 
+  const filteredImageData = useMemo(() => {
+    const sortedImages: EnvironmentImage[] = images
+      ? [...images]
+          .filter(
+            (image): image is EnvironmentImage =>
+              image !== null && image !== undefined,
+          )
+          .sort(sortBasedOnInstalled)
+      : [];
+    if (_.isEmpty(imageSearch)) return sortedImages;
+    const regExp = new RegExp(`${_.escapeRegExp(imageSearch)}`, 'i');
+    return _.filter(sortedImages, (image) => {
+      return _.some(image, (value, key) => {
+        if (key === 'id') return false;
+        if (key === 'installed') return regExp.test(t('environment.Installed'));
+        if (['digest', 'architecture', 'registry'].includes(key))
+          return regExp.test(_.toString(value));
+        if (image.tag && image.name) {
+          const baseVersionMatch = regExp.test(
+            getBaseVersion(getImageFullName(image) || ''),
+          );
+          const baseImagesMatch = getBaseImages(image.tag, image.name).some(
+            (baseImage) => regExp.test(baseImage),
+          );
+          const constraintsMatch = getConstraints(
+            image.tag,
+            image.labels as { key: string; value: string }[],
+          ).some((constraint) => regExp.test(constraint));
+          const customizedMatch =
+            image.tag.indexOf('customized') !== -1
+              ? regExp.test('customized')
+              : false;
+          const langMatch = regExp.test(image.name ? getLang(image.name) : '');
+          const namespaceMatch = regExp.test(
+            getNamespace(getImageFullName(image) || ''),
+          );
+          return (
+            baseVersionMatch ||
+            baseImagesMatch ||
+            constraintsMatch ||
+            langMatch ||
+            namespaceMatch ||
+            customizedMatch
+          );
```

Comment:
If the `filteredImageData` doesn't go through the process of comparing it to processed data rather than image data, the `imageList` will render the wrong data

Reply:
Every `imageSearch` change generates text for filter comparison. It would be better to separate the following concerns:
1. `imageFilterValues`: text for filter comparison when `images` change.
2. `filteredImages`: filtered images based on `imageSearch` and `imageFilterValues`.

Somthing like this:
```
const imageFilterValues = useMemo(()=>{

},[image]);
const filteredImages = useMemo(()=>{
  return _.filter(images, (img, idx)=>{
    // check imageFilterValues[idx] has imageSearch
  })
},[imageFilterValues, imageSearch]);

<Table dataSource={filteredImages}/>
```

Reply:
Thank you for reviewing  I modified it
https://github.com/lablup/backend.ai-webui/pull/2637/commits/3edc4c6983718a23133d91367d15595443cd60c0